scarb/tests/build.rs

```diff
@@ -293,6 +293,42 @@ fn compile_with_incompatible_cairo_version() {
             Cairo version of Scarb: [..]
 
             error: the required Cairo version of each package must match the current Cairo version
+            help: pass `--ignore-cairo-version` to ignore Cairo version mismatch
+        "#});
+}
+
+#[test]
+fn compile_ignore_cairo_version() {
+    let t = TempDir::new().unwrap();
+    t.child("Scarb.toml")
+        .write_str(
+            r#"
+            [package]
+            name = "hello"
+            version = "0.1.0"
+            cairo-version = "33.33.0"
+            "#,
+        )
+        .unwrap();
+    t.child("src/lib.cairo")
+        .write_str("fn example() -> felt252 { 42 }")
+        .unwrap();
+    Scarb::quick_snapbox()
+        .args(["check", "--ignore-cairo-version"])
+        .current_dir(&t)
+        .assert()
+        .success()
+        .stdout_matches(indoc! {r#"
+            warn: the required Cairo version of package hello is not compatible with current version
+            Cairo version required: ^33.33.0
+            Cairo version of Scarb: [..]
+
+            warn: the required Cairo version of package hello is not compatible with current version
+            Cairo version required: ^33.33.0
+            Cairo version of Scarb: [..]
```

Comment:
We previously discussed offline the idea of storing the fact that the warning occurred to avoid duplication. 

However, on the second look, it seems to me a cleaner solution would be to just move the check to `generate_compilation_units`. 

What do you think?